include/libopencm3/stm32/common/usart_common_l0f0.h

```diff
@@ -0,0 +1,337 @@
+#define USART4_RDR			USART_RDR(USART4_BASE)
+
+#define USART_TDR(usart_base)		MMIO8((usart_base) + 0x28)
+#define USART1_TDR			USART_TDR(USART1_BASE)
+#define USART2_TDR			USART_TDR(USART2_BASE)
+#define USART3_TDR			USART_TDR(USART3_BASE)
+#define USART4_TDR			USART_TDR(USART4_BASE)
+
+/*****************************************************************************/
+/* Register values                                                           */
+/*****************************************************************************/
+
+/* USART_CR1 Values ---------------------------------------------------------*/
+
+#define USART_CR1_M1			(1 << 28) /* F07x */
+#define USART_CR1_EOBIE			(1 << 27)
+#define USART_CR1_RTOIE			(1 << 26)
+
+#define USART_CR1_DEAT_SHIFT		21
+#define USART_CR1_DEAT			(0x1F << USART_CR1_DEAT_SHIFT)
+#define USART_CR1_DEAT_VAL(x)		((x) << USART_CR1_DEAT_SHIFT)
+
+#define USART_CR1_DEDT_SHIFT		16
+#define USART_CR1_DEDT			(0x1F << USART_CR1_DEDT_SHIFT)
+#define USART_CR1_DEDT_VAL(x)		((x) << USART_CR1_DEDT_SHIFT)
+
+#define USART_CR1_OVER8			(1 << 15)
+#define USART_CR1_CMIE			(1 << 14)
+#define USART_CR1_MME			(1 << 13)
+#define USART_CR1_M			(1 << 12) /* Obsolete, please use M0 */
+#define USART_CR1_M0			(1 << 12)
+#define USART_CR1_WAKE			(1 << 11)
+#define USART_CR1_PCE			(1 << 10)
+#define USART_CR1_PS			(1 << 9)
+#define USART_CR1_PEIE			(1 << 8)
+#define USART_CR1_TXEIE			(1 << 7)
+#define USART_CR1_TCIE			(1 << 6)
+#define USART_CR1_RXNEIE		(1 << 5)
+#define USART_CR1_IDLEIE		(1 << 4)
+#define USART_CR1_TE			(1 << 3)
+#define USART_CR1_RE			(1 << 2)
+#define USART_CR1_UESM			(1 << 1)
+#define USART_CR1_UE			(1 << 0)
+
+/* USART_CR2 Values ---------------------------------------------------------*/
+
+#define USART_CR2_ADD_SHIFT		24
+#define USART_CR2_ADD			(0xFF << USART_CR2_ADD_SHIFT)
+#define USART_CR2_ADD_VAL(x)		((x) << USART_CR2_ADD_SHIFT)
+
+#define USART_CR2_RTOEN			(1 << 23)
+
+#define USART_CR2_ABRMOD_SHIFT		21
+#define USART_CR2_ABRMOD		(3 << USART_CR2_ABRMOD_SHIFT)
+#define USART_CR2_ABRMOD_STARTBIT	(0 << USART_CR2_ABRMOD_SHIFT)
+#define USART_CR2_ABRMOD_FALLTOFALL	(1 << USART_CR2_ABRMOD_SHIFT)
+
+#define USART_CR2_ABREN			(1 << 20)
+#define USART_CR2_MSBFIRST		(1 << 19)
+#define USART_CR2_DATAINV		(1 << 18)
+#define USART_CR2_TXINV			(1 << 17)
+#define USART_CR2_RXINV			(1 << 16)
+#define USART_CR2_SWAP			(1 << 15)
+#define USART_CR2_LINEN			(1 << 14)
+
+#define USART_CR2_STOP_SHIFT		12
+#define USART_CR2_STOP			(3 << USART_CR2_STOP_SHIFT)
+#define USART_CR2_STOP_1_0BIT		(0 << USART_CR2_STOP_SHIFT)
+#define USART_CR2_STOP_2_0BIT		(2 << USART_CR2_STOP_SHIFT)
+#define USART_CR2_STOP_1_5BIT		(3 << USART_CR2_STOP_SHIFT)
+
+#define USART_CR2_CLKEN			(1 << 11)
+#define USART_CR2_CPOL			(1 << 10)
+#define USART_CR2_CPHA			(1 << 9)
+#define USART_CR2_LBCL			(1 << 8)
+#define USART_CR2_LBIDE			(1 << 6)
+#define USART_CR2_LBDL			(1 << 5)
+#define USART_CR2_ADDM			(1 << 4) /* Obsolete, use ADDM7 */
```

Comment:
same here, but these three little things we can just drop in a later commit I think. no need to update this pull for them.